src/Integration.Vsix/SonarLintTagger/TaggerProvider.cs

```diff
@@ -181,14 +186,34 @@ public void RequestAnalysis(string path, string charset, IEnumerable<AnalysisLan
             // May be called on the UI thread -> unhandled exceptions will crash VS
             try
             {
-                analyzerController.ExecuteAnalysis(path, charset, detectedLanguages, issueConsumer, projectItem);
+                var cancellationToken = GetCancellationToken(path);
+                analyzerController.ExecuteAnalysis(path, charset, detectedLanguages, issueConsumer, projectItem, cancellationToken);
             }
             catch (Exception ex) when (!Microsoft.VisualStudio.ErrorHandler.IsCriticalException(ex))
             {
                 logger.WriteLine($"Analysis error: {ex.ToString()}");
             }
         }
 
+        private CancellationToken GetCancellationToken(string path)
+        {
+            CancellationToken token;
+            lock (analysisJobs)
+            {
+                if (analysisJobs.ContainsKey(path))
+                {
+                    logger.WriteLine($"Cancelled analysis for {path}");
+                    analysisJobs[path].Cancel(throwOnFirstException: false);
+                    analysisJobs[path].Dispose();
```

Comment:
What happens to the cancellation source if the analysis runs to completion? It just stays in the dictionary until the next analysis is triggered for that file?

Reply:
> What happens to the cancellation source if the analysis runs to completion? It just stays in the dictionary until the next analysis is triggered for that file?

Yep. The last analysis for each file will be kept. 
I didn't think it was a problem, but now that you mention it, it will blow up with prolonged work. I will fix it.